src/openms/include/OpenMS/FILTERING/TRANSFORMERS/SpectraMerger.h

```diff
@@ -297,22 +298,76 @@ namespace OpenMS
       exp.sortSpectra();
     }
 
+    /**
+     * @brief check if the first and second mzs might be from the same mass
+     *
+     * @param mz1 the first m/z value
+     * @param mz2 the second m/z value
+     * @param tol_ppm tolerance in ppm
+     * @param max_c maximum possible charge value
+     */
+    bool areMassesMatched(double mz1, double mz2, double tol_ppm, int max_c)
+    {
+      if (mz1 == mz2 || tol_ppm <= 0)
+      {
+        return true;
+      }
+
+      const int min_c = 1;
+      const int max_iso_diff = 5; // maximum charge difference  5 is more than enough
+      const double max_charge_diff_ratio = 3.0; // maximum ratio between charges (large / small charge)
+
+      for (int c1 = min_c; c1 <= max_c; ++c1)
+      {
+        double mass1 = (mz1 - Constants::PROTON_MASS_U) * c1;
+
+        for (int c2 = min_c; c2 <= max_c; ++c2)
+        {
+          if (c1 / c2 > max_charge_diff_ratio)
+          {
+            continue;
+          }
+          if (c2 / c1 > max_charge_diff_ratio)
+          {
+            break;
+          }
+
+          double mass2 = (mz2 - Constants::PROTON_MASS_U) * c2;
+
+          if (fabs(mass1 - mass2) > max_iso_diff)
+          {
+            continue;
+          }
+          for (int i = -max_iso_diff; i <= max_iso_diff; ++i)
+          {
+            if (fabs(mass1 - mass2 + i * Constants::ISOTOPE_MASSDIFF_55K_U) < mass1 * tol_ppm * 1e-6)
+            {
+              return true;
+            }
+          }
+        }
+      }
+      return false;
+    }
+
     /**
      * @brief average over neighbouring spectra
      *
      * @param exp experimental data to be averaged
      * @param average_type averaging type to be used ("gaussian" or "tophat")
+     * @param ms_level targe MS level. If it is -1, ms_level will be determined by ms_level parameter.
```

Comment:
```suggestion
     * @param ms_level target MS level. If it is -1, ms_level will be determined by ms_level parameter.
```

I dont understand what this means: "If it is -1, ms_level will be determined by ms_level parameter" -> it sounds like "X will be determined by X" ?